Just did another test of this (latest version), most modes work fine, but I'm seeing problems with some.

When selecting 'hot' or 'fan' from the HA UI I get the task exceptions errors below (and nothing happens on the AC). When selecting 'heat' from HomeKit, the bottom (invalid value) error shows. All other operation modes seem to work fine.

`2018-10-02 12:59:09 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/helpers/service.py", line 224, in _handle_service_platform_call
    await getattr(entity, func)(**data)
  File "/usr/lib/python3.5/asyncio/futures.py", line 380, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 304, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 293, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/homeassistant/.homeassistant/custom_components/climate/daikin.py", line 239, in set_operation_mode
    self.set({ATTR_OPERATION_MODE: operation_mode})
  File "/home/homeassistant/.homeassistant/custom_components/climate/daikin.py", line 178, in set
    values[daikin_attr] = HA_STATE_TO_DAIKIN[value]
KeyError: 'hot'
2018-10-02 12:59:20 ERROR (MainThread) [homeassistant.core] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "/srv/homeassistant/lib/python3.5/site-packages/homeassistant/helpers/service.py", line 224, in _handle_service_platform_call
    await getattr(entity, func)(**data)
  File "/usr/lib/python3.5/asyncio/futures.py", line 380, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 304, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 293, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/homeassistant/.homeassistant/custom_components/climate/daikin.py", line 239, in set_operation_mode
    self.set({ATTR_OPERATION_MODE: operation_mode})
  File "/home/homeassistant/.homeassistant/custom_components/climate/daikin.py", line 178, in set
    values[daikin_attr] = HA_STATE_TO_DAIKIN[value]
KeyError: 'fan'
2018-10-02 13:00:12 ERROR (Thread-2) [custom_components.climate.daikin] Invalid value operation_mode for heat`

--- a/homeassistant/components/climate/daikin.py
+++ b/homeassistant/components/climate/daikin.py
@@ -84,7 +84,7 @@ class DaikinClimate(ClimateDevice):
         self._api = api
         self._force_refresh = False
         self._list = {
-            ATTR_OPERATION_MODE: list(DAIKIN_TO_HA_STATE),
+            ATTR_OPERATION_MODE: list(HA_STATE_TO_DAIKIN),
             ATTR_FAN_MODE: list(
                 map(
                     str.title,
@@ -143,8 +143,7 @@ class DaikinClimate(ClimateDevice):
         elif key == ATTR_OPERATION_MODE:
             # Daikin can return also internal states auto-1 or auto-7
             # and we need to translate them as AUTO
-            daikin_mode = re.sub(
-                '[^a-z]', '', self._api.device.represent(daikin_attr)[1])
+            daikin_mode = re.sub('[^a-z]', '', self._api.device.represent(daikin_attr)[1])
             
             ha_mode = DAIKIN_TO_HA_STATE.get(daikin_mode)
             value = ha_mode

--- a/homeassistant/components/climate/daikin.py
+++ b/homeassistant/components/climate/daikin.py
@@ -144,7 +144,6 @@ class DaikinClimate(ClimateDevice):
             # Daikin can return also internal states auto-1 or auto-7
             # and we need to translate them as AUTO
             daikin_mode = re.sub('[^a-z]', '', self._api.device.represent(daikin_attr)[1])
-            
             ha_mode = DAIKIN_TO_HA_STATE.get(daikin_mode)
             value = ha_mode
 

--- a/homeassistant/components/climate/daikin.py
+++ b/homeassistant/components/climate/daikin.py
@@ -143,7 +143,8 @@ class DaikinClimate(ClimateDevice):
         elif key == ATTR_OPERATION_MODE:
             # Daikin can return also internal states auto-1 or auto-7
             # and we need to translate them as AUTO
-            daikin_mode = re.sub('[^a-z]', '', self._api.device.represent(daikin_attr)[1])
+            daikin_mode = re.sub('[^a-z]', '',
+               self._api.device.represent(daikin_attr)[1])
             ha_mode = DAIKIN_TO_HA_STATE.get(daikin_mode)
             value = ha_mode
 

